src/pipeline/src/etl.rs

```diff
@@ -32,6 +36,60 @@ pub enum Content {
     Yaml(String),
 }
 
+fn set_processor_keys_index(
```

Comment:
Adds some comments to these functions. How the key and index are mapping like that?